pkg/cri/resource-manager/policy/builtin/topology-aware/resources.go

```diff
@@ -949,6 +949,22 @@ func newRequest(container cache.Container) Request {
 	log.Debug("%s: CPU preferences: cpuType=%s, full=%v, fraction=%v, isolate=%v",
 		container.PrettyName(), cpuType, full, fraction, isolate)
 
+	if full > 0 && fraction > 0 {
+		state := container.GetState()
+		if state != cache.ContainerStateCreating {
+			if !container.HasCPURequestAnnotation() {
+				fraction += 1000 * full
+				full = 0
+				log.Warn("%s: state %d => falling back to shared CPU (%d) allocation",
+					container.PrettyName(), container.GetState(), fraction)
```

Comment:
I think a short note/reasoning comment here too would make future readers happier :) I.e. why are we doing this "hack" for mixed allocations,